### ISSUE TITLE
regrid operation fails when target bounds are completely outside of the element bounds

### ISSUE THREAD

**Issue by stevievb** [ioam/holoviews#2693]
```
import holoviews as hv
import numpy as np
from holoviews.operation.datashader import regrid

target_bounds = (2,2, 3, 3)

ls = np.linspace(0, 10, 200)
lsy = np.linspace(0, 10, 400)
xx, yy = np.meshgrid(ls, lsy)
bounds = (-1,-1,1,1)
img = hv.Image(np.sin(xx)*np.cos(yy), bounds=bounds)
target = hv.Image(np.zeros((200, 200)), bounds=target_bounds)
regridded_op = regrid.instance(target=target, expand=True, dynamic=False)
regridded_op(img)
```
```
~/anaconda3/envs/bli/lib/python3.6/site-packages/datashader/resampling.py in _get_out(out, src, shape)
    191 def _get_out(out, src, shape):
    192     if out is None:
--> 193         return np.zeros(shape, dtype=src.dtype)
    194     else:
    195         if out.shape != shape:

ValueError: negative dimensions are not allowed
```

**Comment by jbednar** [comment]
Hmm. Not sure if this is best caught at the datashader or the holoviews level.

**Comment by philippjfr** [comment]
Right, I suggested filing the issue in holoviews first, so we can triage it and work out where to apply the fix.

**Comment by jbednar** [comment]
Hopefully the fix works best in datashader, as that's about to be released!

**Comment by philippjfr** [comment]
>Hopefully the fix works best in datashader, as that's about to be released!

Sure I can have a look at that tonight or tomorrow.

**Comment by philippjfr** [comment]
Apparently I already fixed this in datashader here https://github.com/bokeh/datashader/pull/573 and the fix is in v0.6.6 which you can install with:

    conda install -c pyviz datashader
